shared/src/balancer/event_handler.rs

```diff
@@ -63,29 +63,33 @@ pub struct WeightedPoolBuilder {
 }
 
 impl WeightedPoolBuilder {
-    fn into_pool(self) -> Option<WeightedPool> {
-        if let (Some(pool_registration), Some(tokens_registration)) =
-            (self.pool_registration, self.tokens_registration)
-        {
-            return Some(WeightedPool {
+    fn into_pool(self) -> Result<RegisteredPool> {
+        if let (Some(pool_registration), Some(tokens_registration)) = (
+            self.pool_registration.clone(),
```

Comment:
why is this extra clone necessary?

Reply:
This is because of the use of self in the error log.